base_layer/core/src/chain_storage/db_transaction.rs

```diff
@@ -211,11 +202,19 @@ impl DbTransaction {
         self
     }
 
-    pub fn set_best_block(&mut self, height: u64, hash: HashOutput, accumulated_difficulty: u128) -> &mut Self {
+    pub fn set_best_block(
+        &mut self,
+        height: u64,
+        hash: HashOutput,
+        accumulated_difficulty: u128,
+        from: HashOutput,
```

Comment:
I still think `from` does not give enough meaning, maybe `expected_prev_best_block` (I know it's long, but the reason for the parameter is clear) - then comment on this function

```rust
/// Sets the best block metadata. 
/// This operation will fail if the header hash was not previously added and if the `expected_prev_best_block` hash 
/// does not match the current best block. This ensures that a data race results in an error and is rolled back.
```